js-api-spec/node-package-importer.node.test.ts

```diff
@@ -0,0 +1,681 @@
+            });
+            expect(result.css).toEqualIgnoringWhitespace('a {b: c;}');
+          },
+          {changeEntryPoint: 'index.js'}
+        );
+      }));
+    it('resolves from secondary @use pkg root', () =>
+      sandbox(dir => {
+        dir.write({
+          'node_modules/bah/index.scss': '@use "pkg:bar";',
+          'node_modules/bah/package.json': manifestBuilder(),
+          'node_modules/bar/index.scss': 'a {b: c}',
+          'node_modules/bar/package.json': manifestBuilder(),
+        });
+        dir.chdir(
+          () => {
+            const result = compileString('@use "pkg:bah";', {
+              importers: [nodePackageImporter],
+            });
+            expect(result.css).toEqualIgnoringWhitespace('a {b: c;}');
+          },
+          {changeEntryPoint: 'index.js'}
+        );
+      }));
+    it('resolves most proximate node_module', () =>
+      sandbox(dir => {
+        dir.write({
+          'node_modules/bah/index.scss': '@use "pkg:bar";',
+          'node_modules/bah/package.json': manifestBuilder(),
+          'node_modules/bar/index.scss': 'e {from: root}',
+          'node_modules/bar/package.json': manifestBuilder(),
+          'node_modules/bah/node_modules/bar/index.scss': 'a {b: c}',
+          'node_modules/bah/node_modules/bar/package.json': manifestBuilder(),
+        });
+        dir.chdir(
+          () => {
+            const result = compileString('@use "pkg:bah";', {
+              importers: [nodePackageImporter],
+            });
+            expect(result.css).toEqualIgnoringWhitespace('a {b: c;}');
+          },
+          {changeEntryPoint: 'index.js'}
+        );
+      }));
+    it('resolves sub node_module', () =>
+      sandbox(dir => {
+        dir.write({
+          'node_modules/bah/index.scss': '@use "pkg:bar";',
+          'node_modules/bah/package.json': manifestBuilder(),
+          'node_modules/bah/node_modules/bar/index.scss': 'a {b: c}',
+          'node_modules/bah/node_modules/bar/package.json': manifestBuilder(),
+        });
+        dir.chdir(
+          () => {
+            const result = compileString('@use "pkg:bah";', {
+              importers: [nodePackageImporter],
+            });
+            expect(result.css).toEqualIgnoringWhitespace('a {b: c;}');
+          },
+          {changeEntryPoint: 'index.js'}
+        );
+      }));
+    it('resolves node_module above cwd', () =>
+      sandbox(dir => {
+        dir.write({
+          'node_modules/bar/index.scss': 'a {b: c}',
+          'node_modules/bar/package.json': manifestBuilder(),
+        });
+        dir.chdir(
+          () => {
+            const result = compileString('@use "pkg:bar";', {
+              importers: [nodePackageImporter],
+            });
+            return expect(result.css).toEqualIgnoringWhitespace('a {b: c;}');
+          },
+          {changeEntryPoint: 'deeply/nested/file/index.js'}
+        );
+      }));
+    it('continues if no match found', () => {
```

Comment:
It's odd that this test name says "continues" but it's actually verifying that the test fails.

Reply:
I think a better way to do it would be to pass in a custom importer and verifying that that importer is passed a `pkg:` URL when the Node importer can't find anything.